### PR TITLE
Only do auth check for tables in TiDB

### PR DESCRIPTION
**Pull request by TrafalgarRicardoLu**
### What problem does this PR solve? 
Only do auth check for tables in TiDB.

### What is changed and how it works?
We will check the catalog name of operation. If it doesn't use `tidb_catalog`, we won't do auth check for it.

### Check List 

Tests 
 - Unit test
 - Integration test

### CODE REVIEW

**Comment by shiyuhang0** [comment]
It's better to add some test

**Review by xuanyu66 — CHANGES_REQUESTED** [review]
Add some IT. https://github.com/pingcap/tispark/blob/master/core/src/test/scala/com/pingcap/tispark/auth/TiAuthIntegrationSuite.scala

File: core/src/test/scala/com/pingcap/tispark/auth/TiAuthIntegrationSuite.scala [hunk 1]
```diff
@@ -79,10 +78,20 @@ class TiAuthIntegrationSuite extends SharedSQLContext {
     tidbStmt.execute(s"DROP DATABASE IF EXISTS `$dummyDatabase`")
     super.afterAll()
     _isAuthEnabled = false
+    _isHiveEnabled = false
     TiAuthorization.enableAuth = false
   }
 
+  test("Operator on hive table should pass") {
```
Comment (xuanyu66) [inline core/src/test/scala/com/pingcap/tispark/auth/TiAuthIntegrationSuite.scala:85]: move methods before afterAll()

**Review by shiyuhang0 — APPROVED** [review]

**Review by Daemonxiao — APPROVED** [review]

**Review by xuanyu66 — APPROVED** [review]

**Pull request merged**